### ISSUE TITLE
Overfit when Shuffle is True

### ISSUE THREAD

**Issue by surya-narayanan** [PyTorchLightning/pytorch-lightning#7656]
## 🐛 Bug

When overfit is True in the Trainer and Shuffle is True in the Data Loader, I get the following Warning 

`/opt/conda/envs/pl124_surya/lib/python3.8/site-packages/pytorch_lightning/utilities/distributed.py:52: UserWarning: You requested to overfit but enabled test/val dataloader shuffling. We are turning it off for you.
`

I am not sure which of overfit or shuffle you are turning off, when you say "We are turning it off for you". 

### To Reproduce

https://colab.research.google.com/drive/1MhFuW8g0_MyqL6aOWZZJC1y3dKU_Ngrn?usp=sharing

### Expected behavior

Tell me which of overfit / shuffle you are turning off

### Environment

* CUDA:
	- GPU:
		- Tesla T4
	- available:         True
	- version:           10.1
* Packages:
	- numpy:             1.19.5
	- pyTorch_debug:     False
	- pyTorch_version:   1.8.1+cu101
	- pytorch-lightning: 1.3.2
	- tqdm:              4.41.1
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.7.10
	- version:           #1 SMP Tue Apr 20 19:55:43 PDT 2021

**Comment by Lucklyric** [comment]
Dear @surya-narayanan , accordind to:
https://github.com/PyTorchLightning/pytorch-lightning/blob/6e6e29af49df7bf0e27e8197ccd3b9c3e3adaba6/pytorch_lightning/trainer/data_loading.py#L237-L247
It turns the training dataset shuffling off.
